src/bootstrap/doc.rs

```diff
@@ -729,14 +735,22 @@ macro_rules! tool_doc {
     }
 }
 
-tool_doc!(Rustdoc, "rustdoc-tool", "src/tools/rustdoc", ["rustdoc", "rustdoc-json-types"]);
+tool_doc!(
+    Rustdoc,
+    "rustdoc-tool",
+    "src/tools/rustdoc",
+    ["rustdoc", "rustdoc-json-types"],
+    in_tree = true
+);
 tool_doc!(
     Rustfmt,
     "rustfmt-nightly",
     "src/tools/rustfmt",
     ["rustfmt-nightly", "rustfmt-config_proc_macro"],
+    in_tree = true
```

Comment:
FWIW I think `in_tree: true` might be more Rust-style syntax here.

Reply:
> in_tree: true feels more like when you are constructing a structure,

Yes, exactly. That is an argument in favor of `:`, isn't it?

`in_tree = true` looks like a `let`-binding, but that would mean the name is *our* choice, which it is not.

Reply:
This is consistent with the macro in `tool.rs`, I'd prefer to keep it as-is.
https://github.com/rust-lang/rust/blob/c62c482b82f4ea5d3d6b87d5f85b87f2fb0dc4ab/src/bootstrap/tool.rs#L693

Reply:
Ah, fair. Sorry for the noise.

Reply:
> so what do I do ? roll back

I'd appreciate that, thanks. You'll need to rebase this PR over the miri update in any case.

Reply:
> > so what do I do ? roll back
> 
> I'd appreciate that, thanks. You'll need to rebase this PR over the miri update in any case.

For some reason, git refuses to push the rebased branch

```
  1 src/bootstrap/ git … push -v --force fork refs/heads/miri-docs\:refs/heads/miri-docs
Pushing to github.com:InfRandomness/rust.git
Running pre-push script '/home/infrandomness/Documents/Dev/rust/x.py test tidy --bless'
Updating only changed submodules
Updating submodule src/tools/miri
remote: Total 0 (delta 0), reused 0 (delta 0), pack-reused 0        
fatal: 'fork' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: Fetched in submodule path 'src/tools/miri', but it did not contain 3361eabf3882578207a483f2cee631646e80eabc. Direct fetching of that commit failed.
Failed updating submodule. This is probably due to uncommitted local changes.
Either stash the changes by running "git stash" within the submodule's
directory, reset them by running "git reset --hard", or commit them.
To reset all submodules' changes run "git submodule foreach --recursive git reset --hard".
Build completed unsuccessfully in 0:00:00
error: failed to push some refs to 'github.com:InfRandomness/rust.git'
```

Reply:
This is a bug in git - try running `git submodule deinit -f src/tools/miri` and seeing if it fixes the issue.
https://rust-lang.zulipchat.com/#narrow/stream/182449-t-compiler.2Fhelp/topic/More.20submodule.20problems

Reply:
> `git submodule deinit -f src/tools/miri`

It worked but I had to remove the pre-push hook that invoked x.py because it reinitialized the submodule everytime